src/main/java/org/openrewrite/java/testing/junit5/RemoveTryCatchBlocksFromUnitTests.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.junit5;
+
+import org.openrewrite.*;
+import org.openrewrite.java.*;
+import org.openrewrite.java.tree.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RemoveTryCatchBlocksFromUnitTests extends Recipe {
+    private static final MethodMatcher ASSERT_FAIL_MATCHER = new MethodMatcher("org.junit.Assert fail(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "Unit test should throw exceptions instead of using `try-catch` blocks";
+    }
+
+    @Override
+    public String getDescription() {
+        return "When the code under test in a unit test throws an exception, the test itself fails. " +
+                "Therefore, there is no need to surround the tested code with a try-catch.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-3658");
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RemoveTryCatchBlocksFromUnitsTestsVisitor();
+    }
+
+    private static class RemoveTryCatchBlocksFromUnitsTestsVisitor extends JavaVisitor<ExecutionContext> {
+        @Override
+        public J visitTry(J.Try try_, ExecutionContext ctx) {
+            if (try_.getCatches().get(0).getBody().getStatements().stream().noneMatch(
```

Comment:
- [ ] We seem to imply here that we only ever look at the first `catch` block;
  I'd appreciate a test that uses two catch blocks, and appropriate handling (skip?)
- [ ] Have you given any thought to try-with-resources blocks? We would not want to remove those.
  Might be good to add in a test that uses `try (Foo bar) { ... } catch (Exception e) { fail(); }`

Reply:
Yes I am not sure how we should handle try statements with multiple catch blocks. Did you see Knut's comment about using AssertJ instead? What are your thoughts on that?

Reply:
There might be value in having a separate recipe for AssertJ, but not everyone is ready or willing to use that just yet. To me it then makes the most sense to check here that there is:
1. only one catch block, such that we know it's safe to apply this recipe
2. only one statement in the catch block, which is a `fail()`, with no or a simple String argument. We would not want to convert for instance `fail(cleanUpAndReturnMessage())` might still have side effects that we don't want to remove.

Reply:
Hmm; considering number two above: `Assert.fail(e.getMessage())` should be fine.. It's not easy thinking of each possible scenario 😅

Reply:
> There might be value in having a separate recipe for AssertJ, but not everyone is ready or willing to use that just yet. To me it then makes the most sense to check here that there is:
> 
>     1. only one catch block, such that we know it's safe to apply this recipe
> 
>     2. only one statement in the catch block, which is a `fail()`, with no or a simple String argument. We would not want to convert for instance `fail(cleanUpAndReturnMessage())` might still have side effects that we don't want to remove.

Latest commit addresses this